File: JAVA_course/webAppBatis/src/board/Dao.java
package board;

import java.util.ArrayList;

import org.apache.ibatis.annotations.Delete;
import org.apache.ibatis.annotations.Insert;
import org.apache.ibatis.annotations.Param;
import org.apache.ibatis.annotations.Select;
import org.apache.ibatis.annotations.Update;

public interface Dao {
	
	@Insert("insert into board(writer, w_date, title, content) values(#{writer},sysdate(),#{title},#{content})")
	void insert(Board b); // 현재날짜로 저장(입력x)
	
	@Select("select * from board where num=#{num}")
	Board selectByNum(@Param("num") int num); // 글 번호로 검색
	
	@Select("select * from board where writer=#{writer}")
	ArrayList<Board> selectByWriter(@Param("writer") String writer); // 작성자로 검색
	
	@Select("select * from board where title like #{title}")
	ArrayList<Board> selectByTitle(@Param("title") String title);  // 제목으로 검색 like활용
	
	@Select("select * from board order by num")
	ArrayList<Board> selectAll(); // 전체검색
	
	@Update("update board set w_date=sysdate(), title=#{title}, content=#{content} where num=#{num}")
	void update(Board b); // 글 수정, 글 번호로 찾아서 제목과 내용 변경(날짜는 현재날짜로)
	
	@Delete("delete from board where num=#{num}")
	void delete(int num); // 글 삭제
}

File: Mini_Project/board/article/dao.py
import pymysql
import board.article.vo as art_vo

class DAO:

    def connect(self):
        return pymysql.connect(host='localhost', user='testuser', password='testpw',
                       db='testdb', charset='utf8')

    def insert(self,m):
        sql = 'insert into board(writer,w_date,title,content) values(%s,now(),%s,%s)'
        d = (m.writer, m.title, m.content)
        conn = self.connect()
        cursor = conn.cursor()
        cursor.execute(sql,d)
        conn.commit()
        conn.close()

    def selectAll(self):
        conn = self.connect()
        sql = 'select * from board'
        cursor = conn.cursor()
        cursor.execute(sql)
        datas = []

        for row in cursor:
            datas.append(art_vo.Board(row[0], row[1], row[2], row[3], row[4]))
        conn.close()
        return datas

    def numSelect(self,num):
        conn = self.connect()
        sql = 'select * from board where num = %s'
        cursor = conn.cursor()
        cursor.execute(sql, num)
        row = cursor.fetchone()
        conn.close()
        if row != None:
            return art_vo.Board(row[0], row[1], row[2], row[3], row[4])

    def titleSelect(self,title):
        conn = self.connect()
        sql = 'select * from board where title like \'%' + title + '%\''
        cursor = conn.cursor()
        cursor.execute(sql)
        datas = []
        for row in cursor:
            datas.append(art_vo.Board(row[0], row[1], row[2], row[3], row[4]))
        conn.close()
        return datas

    def writerSelect(self,writer):
        conn = self.connect()
        sql = 'select * from board where writer = %s'
        cursor = conn.cursor()
        cursor.execute(sql, writer)
        datas = []
        for row in cursor:
            datas.append(art_vo.Board(row[0], row[1], row[2], row[3], row[4]))
        conn.close()
        return datas


    def update(self,m): # 아직
        sql = 'update board set w_date = now() , title=%s, content=%s where num=%s'
        conn = self.connect()
        cursor = conn.cursor()
        d = (m.title, m.content, m.num)
        cursor.execute(sql, d)
        conn.commit()
        conn.close()

    def delete(self,num):
        conn = self.connect()
        sql = 'delete from board where num=%s'
        cursor = conn.cursor()  # 사용할 커서 객체 생성
        cursor.execute(sql, num)
        conn.commit()  # 쓰기 완료
        conn.close()